pingen/pingen_document.py

```diff
@@ -331,95 +307,83 @@ def _update_post_infos(self, cr, uid, document, pingen, context=None):
-            cr, uid, [('name', '=', post_infos['currency'])], context=context)
-        country_ids = self.pool.get('res.country').search(
-            cr, uid, [('code', '=', post_infos['country'])], context=context)
-
+        currency_ids = self.env['res.currency'].search(
+            [('name', '=', post_infos['currency'])])
+        country_ids = self.env['res.country'].search(
+            [('code', '=', post_infos['country'])])
         send_date = pingen_datetime_to_utc(post_infos['date'])
-
         vals = {
             'post_status': POST_SENDING_STATUS[post_infos['status']],
             'cost': post_infos['cost'],
             'currency_id': currency_ids[0] if currency_ids else False,
             'parsed_address': post_infos['address'],
             'country_id': country_ids[0] if country_ids else False,
-            'send_date': send_date.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT),
+            'send_date': send_date.strftime(
+                tools.DEFAULT_SERVER_DATETIME_FORMAT),
             'pages': post_infos['pages'],
             'last_error_message': False,
             }
         if pingen.is_posted(post_infos):
             vals['state'] = 'sent'
-
-        document.write(vals, context=context)
+        document.write(vals)
         _logger.info('Pingen Document %s: status updated' % document.id)
 
-    def _update_post_infos_cron(self, cr, uid, ids, context=None):
-        """ Update the informations from pingen of a document in the Sendcenter
-
+    def _update_post_infos_cron(self):
+        """ Update the informations from pingen of a
+        document in the Sendcenter
         Intended to be used in a cron.
-
         Commit after each record
-
-        Do not raise errors, only skip the update of the record.
-        """
-        if not ids:
-            ids = self.search(
-                cr, uid,
-                [('state', '=', 'sendcenter')],
-                context=context)
-
-        with closing(pooler.get_db(cr.dbname).cursor()) as loc_cr, \
-                self._get_pingen_session(cr, uid, context=context) as session:
-            for document in self.browse(loc_cr, uid, ids, context=context):
-                try:
-                    self._update_post_infos(
-                        loc_cr, uid, document, pingen=session, context=context)
-                except (ConnectionError, APIError):
-                    # will be retried the next time
-                    # In any case, the error has been logged by _update_post_infos
-                    loc_cr.rollback()
-                except:
-                    _logger.error('Unexcepted error in pingen cron')
-                    loc_cr.rollback()
-                    raise
-                else:
-                    loc_cr.commit()
+        Do not raise errors, only skip the update of the record."""
+        cr = registry(self._cr.dbname).cursor()
+        self = self.with_env(self.env(cr=cr))
+        session = self._get_pingen_session()
```

Comment:
If you have an unexpected exception this cursor won't be closed as you removed the use of this cursor as a context manager. Some for the pingen session. Please keep the context manager.


``` python
with self.pool.cursor() as cr, self._get_pingen_session() as session:
    self = self.with_env(self.env(cr=cr))
```

Might be what you are looking for.